### ISSUE TITLE
Can huggingface Trainer features be used in setfit?

### ISSUE THREAD

**Issue by AuravPadhi** [huggingface/setfit#158]
How can I use the following features in setfit

- Model parallelism
- Loading best checkpoint after training
- wandb integration like in the hf Trainer class

**Comment by lewtun** [comment]
Hi @AuravPadhi the `SetFitTrainer` is a rather lean version of the one in `transformers`. Regarding model parallelism, is there a particular use case you have in mind? In our experiments, we found 1 GPU was sufficient in the low-data regime

For the other two features, a pull request would be more than welcome! We could include support for callbacks, which would enable the WandB integration to be very simple :)

**Comment by AnshulP10** [comment]
I have a dataset of 80-90 classes with ~20 samples each and based on the hyperparameter search I found that the accuracy is best for 4 epochs so would need multi-gpu support. And is there a guideline for selecting the number of iterations if we have a large number of classes. I am currently using the standard 20

**Comment by lewtun** [comment]
Thanks for sharing @AnshulP10 ! I believe @PhilipMay has run experiments at a similar scale, so perhaps they can share what iteration number worked well for them

**Comment by PhilipMay** [comment]
Here is what I found for iterations in optuna for my usecase:

<img width="247" alt="image" src="https://user-images.githubusercontent.com/229382/200833840-04de7d2f-17d3-4cad-82af-5c2ef7eeb1eb.png">

**Comment by AnshulP10** [comment]
Thanks a lot @PhilipMay ! 
Could you tell me how many labels you have and whether it was multilabel or multiclass?
And are all these experiments for one epoch?

**Comment by PhilipMay** [comment]
I have multi class:

- 3660 train samples
- minority classes in train set has 20 samples
- majority classes in train set has 183 samples
- I am doing 10 fold cross validation on the train set
- the mean of the first 4 folds is the result of my hyperparameter optimization

**Comment by tomaarsen** [comment]
I believe this has been answered now, so I'll close this :)
